### PR TITLE
feat: Support graphql-kotlin-bom

### PR DESCRIPTION
**Pull request by mykevinjung**
### :pencil: Description

Adding a support on `graphql-kotlin-bom`

Generated BOM looks like this:

```
<?xml version="1.0" encoding="UTF-8"?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <!-- This module was also published with a richer model, Gradle metadata,  -->
  <!-- which should be used instead. Do not delete the following line which  -->
  <!-- is to indicate to Gradle or any Gradle module metadata file consumer  -->
  <!-- that they should prefer consuming it instead. -->
  <!-- do_not_remove: published-with-gradle-metadata -->
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.expediagroup</groupId>
  <artifactId>graphql-kotlin-bom</artifactId>
  <version>8.0.0-SNAPSHOT</version>
  <packaging>pom</packaging>
  <name>graphql-kotlin-bom</name>
  <description>BOM (Bill Of Materials) for graphql-kotlin</description>
  <url>https://github.com/ExpediaGroup/graphql-kotlin</url>
  <organization>
    <name>https://www.expediagroup.com/</name>
  </organization>
  <licenses>
    <license>
      <name>The Apache Software License, Version 2.0</name>
      <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
    </license>
  </licenses>
  <developers>
    <developer>
      <name>Expedia Group Committers</name>
      <email>oss@expediagroup.com</email>
      <organization>Expedia Group</organization>
      <organizationUrl>https://www.expediagroup.com/</organizationUrl>
    </developer>
  </developers>
  <scm>
    <connection>scm:git:git://github.com/ExpediaGroup/graphql-kotlin.git</connection>
    <developerConnection>scm:git:git://github.com/ExpediaGroup/graphql-kotlin.git</developerConnection>
    <url>https://github.com/ExpediaGroup/graphql-kotlin</url>
  </scm>
  <dependencies>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-automatic-persisted-queries</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-client</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-client-generator</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-client-jackson</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-client-serialization</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-dataloader</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-dataloader-instrumentation</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-federated-hooks-provider</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-federation</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-graalvm-metadata-generator</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-gradle-plugin</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-hooks-provider</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-ktor-client</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-ktor-server</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-maven-plugin</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-schema-generator</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-sdl-generator</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-server</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-spring-client</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.expediagroup</groupId>
      <artifactId>graphql-kotlin-spring-server</artifactId>
      <version>8.0.0-SNAPSHOT</version>
      <scope>compile</scope>
    </dependency>
  </dependencies>
</project>
```

### CODE REVIEW

File: bom/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                        email.set("oss@expediagroup.com")
+                        organization.set("Expedia Group")
+                        organizationUrl.set("https://www.expediagroup.com/")
+                    }
+                }
+                scm {
+                    connection.set("scm:git:git://github.com/ExpediaGroup/graphql-kotlin.git")
+                    developerConnection.set("scm:git:git://github.com/ExpediaGroup/graphql-kotlin.git")
+                    url.set("https://github.com/ExpediaGroup/graphql-kotlin")
+                }
```
Comment (mykevinjung) [inline bom/build.gradle.kts:51]: These are defined in [com.expediagroup.graphql.conventions.gradle.kts](https://github.com/ExpediaGroup/graphql-kotlin/blob/d038e2e4aa8229e53e24c9046aa16c0c1a2378a6/buildSrc/src/main/kotlin/com.expediagroup.graphql.conventions.gradle.kts#L78-L112) and other java modules are using `id("com.expediagroup.graphql.conventions")` plugin.  But that plugin is a common module for java libraries, not applicable for this BOM.  copy-create with `pom` packaging.

**Comment by dariuszkuc** [comment]
Whats the point/value of the platform jar/bom for this OSS project?
- for Spring based servers -> you just need to import `spring-server` and it will transitively include all the other required modules
- for Ktor based servers -> you just depend on the `ktor-server` to pull in the deps
- plugins management doesn't really work with bom?
- guess for clients it might be useful?

**Comment by mykevinjung** [comment]
@dariuszkuc Applications may need multiple modules.  For instance, GraphQL server apps, at least at EG, use server, client for integration test and maven-plugin to generate SDL in common.  Also if a team develops a parent POM for other devs, then for instance, server parent POM may include both ktor and spring, and client parent POM may include ktor, spring, serialization or jackson so that app developers can choose whichever server/client module they want.  These can be wrapped up in the parent POM but BOM seems to be a better place.  Also I thought it's a good practice to provide a BOM in general when there are many modules in the project, 20+ modules here.

**Comment by dariuszkuc** [comment]
If you are keen on creating the platform/bom then I'd suggest to also make it more useful to also specify some required dependency versions (e.g. `graphql-kotlin` v8 depends on `graphql-java` v22 etc)

**Comment by mykevinjung** [comment]
It seems that the usage of `graphql-kotlin` requires just one module most time.  BOM doesn't seem to be beneficial in this case although there are many modules.  closing

**Pull request closed without merging**